actors/hierarchical_sca/src/lib.rs

```diff
@@ -629,6 +635,245 @@ impl Actor {
 
         Ok(())
     }
+
+    /// Initializes an atomic execution to be orchestrated by the current subnet.
+    /// This method verifies that the execution is being orchestrated by the right subnet
+    /// and that its semantics and inputs are correct.
+    fn init_atomic_exec<BS, RT>(
+        rt: &mut RT,
+        params: AtomicExecParams,
+    ) -> Result<LockedOutput, ActorError>
+    where
+        BS: Blockstore,
+        RT: Runtime<BS>,
+    {
+        rt.validate_immediate_caller_type(CALLER_TYPES_SIGNABLE.iter())?;
+
+        // translate inputs into id addresses for the subnet.
+        let mut params = params;
+        params.input_into_ids(rt).map_err(|e| {
+            e.downcast_default(
+                ExitCode::USR_ILLEGAL_ARGUMENT,
+                "error translating execution input addresses to IDs",
+            )
+        })?;
+
+        // get cid for atomic execution
+        let cid = params.cid().map_err(|e| {
+            e.downcast_default(ExitCode::USR_ILLEGAL_ARGUMENT, "error computing Cid for params")
+        })?;
+
+        rt.transaction(|st: &mut State, rt| {
+        match st.get_atomic_exec(rt.store(), &cid.into()).map_err(|e| {
+            e.downcast_default(
+                ExitCode::USR_ILLEGAL_ARGUMENT,
+                "error translating execution input addresses to IDs",
+            )
+        })? {
+            Some(_) => {
+                return Err(actor_error!(
+                    illegal_argument,
+                    format!("execution with cid {} already exists", &cid)
+                ));
+            }
+            None => {
+                // check if exec has correct number of inputs and messages.
+                if params.msgs.len() == 0 || params.inputs.len() < 2 {
+                    return Err(actor_error!(
+                        illegal_argument,
+                        "wrong number of messages or inputs provided for execution"
+                    ));
+                }
```

Comment:
```suggestion
                // check if exec has correct number of inputs and messages.
                if params.inputs.len() < 2 {
                    return Err(actor_error!(
                        illegal_argument,
                        "wrong number of messages or inputs provided for execution"
                    ));
                }
```
It looks redundant. Maybe the comment or the error message should say what it expected. 
As I understand "input" needs to correspond to the number of participants, which should be at least 2.

Reply:
Oh it's `msgs` not `input`, sorry :see_no_evil: